packages/e2e-test-utils/src/customFonts.js

```diff
@@ -35,20 +79,33 @@ export const addCustomFont = async (fontUrl) => {
 
   await page.keyboard.type(fontUrl);
   await expect(page).toClick('button', { text: 'Add Font' });
-  // TODO(#10878): Target font by font name.
-  await page.waitForSelector('[aria-label="Remove font"]');
+  await expect(page).toMatch(fontUrl);
 };
 
 /**
  * Remove a custom font from the settings page.
  *
+ * @param {string} fontName fontName to delete
  * @return {Promise<void>}
  */
-export const removeCustomFont = async () => {
-  await visitSettings();
-  // TODO(#10878): Target font by font name.
-  await page.hover('[aria-label="Remove font"]');
-  await page.click('[aria-label="Remove font"]');
+export const removeCustomFont = async (fontName) => {
+  const numChildren = await page.evaluate(() => {
+    return document.querySelector('div[role=listbox]')?.children?.length;
+  });
+
+  const selector = `[aria-label="Delete ${fontName}"]`;
+  await page.waitForSelector(selector);
+  await page.focus(selector);
+  await page.click(selector);
   await page.waitForSelector('[role="dialog"]');
   await expect(page).toClick('button', { text: 'Delete Font' });
+  await page.waitForFunction(
+    (originalNumChildren) => {
+      const count =
+        document.querySelector('div[role=listbox]')?.children?.length;
+      return !count || count === originalNumChildren - 1 ? true : false;
+    },
+    { polling: 300 },
```

Comment:
If the latest commit works I'll see about removing the polling

Reply:
Removed https://github.com/GoogleForCreators/web-stories-wp/pull/10918/commits/f8edcdff1d8c35ecf14211493a63b6699b27c05a